screens/PokemonList.tsx

```diff
@@ -1,28 +1,20 @@
 import React, { useEffect, useState } from 'react';
 import { StyleSheet, Text, View, FlatList, Dimensions } from 'react-native';
 import { getAllPokemons } from '../api/pokemon';
+import { Pokemon } from '../api/pokemon';
 
 const keyExtractor = ({ name }: { name: string }): string => name;
 
 export default function PokemonList() {
-  interface Pokemon {
-    name: string;
-    weight: number;
-    height: number;
-    base_experience: number;
-    sprites: [object];
-    url: string;
-  }
-
   const step = 20;
   const [batchStart, setBatchStart] = useState(0);
   const [loading, setLoading] = useState(true);
-  const [pokemons, setPokemons] = useState([]);
+  const [pokemons, setPokemons] = useState<ReadonlyArray<Pokemon> | null>(null);
```

Comment:
if you change Pokemons to be an empty list again then I think it should be fine (and always use `[...pokemons, ...allPokemons]`). Why have you changed it to null? (it's also fine, but it forces you to check if `pokemons` are nonnull, which is annoying)